target/linux/octeon/patches-6.6/100-mips_image_cmdline_hack.patch

```diff
@@ -0,0 +1,38 @@
+From: John Crispin <john@phrozen.org>
+Subject: hack: kernel: add generic image_cmdline hack to MIPS targets
+
+lede-commit: d59f5b3a987a48508257a0ddbaeadc7909f9f976
+Signed-off-by: Gabor Juhos <juhosg@openwrt.org>
+---
+ arch/mips/Kconfig       | 4 ++++
+ arch/mips/kernel/head.S | 6 ++++++
+ 2 files changed, 10 insertions(+)
+
+--- a/arch/mips/Kconfig
++++ b/arch/mips/Kconfig
+@@ -1090,6 +1090,10 @@ config MIPS_MSC
+ config SYNC_R4K
+ 	bool
+ 
++config IMAGE_CMDLINE_HACK
```

Comment:
As we are using Device Tree on this platform, why do we still need this hack?
Doesn't `/chosen/bootargs` work just as fine?

Reply:
Oh, I wish. There are Octeon subtargets for which we don't have a device tree at all (e.g. the Itus Shield), it's inherited from the bootloader. Unless we have device trees for all devices, I'm not seeing a trivial way of getting rid of this hack. It's **extremely** desirable, however, since Octeon is the last target requiring this.